web/server/routers/account.ts

```diff
@@ -1,17 +1,57 @@
 import { TRPCError } from "@trpc/server";
 import { z } from "zod";
-import { removeAccount } from "../../db/actions/Account";
+import {
+  findAccount,
+  removeAccount,
+  updateAccount,
+} from "../../db/actions/Account";
 import { updateUser } from "../../db/actions/User";
 import Account from "../../db/models/Account";
 import { router, protectedProcedure } from "../trpc";
 
+const emailInput = z.object({
+  email: z.string().email("Invalid email provided"),
+});
+
 export const accountRouter = router({
+  modify: protectedProcedure
+    .input(emailInput)
+    .mutation(async ({ ctx, input }) => {
+      const email = input.email;
+      if (ctx.session?.email === email)
+        throw new TRPCError({
+          message: "Unauthorized - Cannot modify own account",
+          code: "UNAUTHORIZED",
+        });
+
+      const account = await findAccount(email);
```

Comment:
Use findOneAndUpdate and see if the returned value is `null` to check whether the account exists in the collection. You can modify the procedure input to take in a boolean field for the admin to pass into the update object for mongoose (this was an oversight on my end - my fault!!). You can assume that the frontend will pass in the email and new admin field. Should result in max 3 db calls to 2 (updateAccount and updateUser, remove findAccount).

Reply:
Just pushed a fix with the above changes (force pushed after squashing mixup commits)